### PR TITLE
Feature/e2e login through api

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
#557 

Cypress tests now log in directly through the API.
Logging in through the API is (with [this branch](https://github.com/fecgov/fecfile-web-api/pull/228)) a one-step process.
The local login form is hidden by default and becomes available after checking for its availability with the API.

New unit tests covering this functionality have been added.

### CODE REVIEW

File: front-end/cypress/support/commands.ts [hunk 1]
```diff
@@ -78,6 +127,7 @@ export function logout() {
 
 function retrieveAuthToken() {
   const storedData = localStorage.getItem('fecfile_online_userLoginData');
+  console.log(storedData);
```
Comment (dheitzer) [inline front-end/cypress/support/commands.ts]: maybe delete this?

File: front-end/cypress/support/commands.ts [hunk 2]
```diff
@@ -26,6 +51,30 @@ function getLoginIntervalString(sessionDur: number): string {
...
+    method: 'POST',
+    url: 'http://localhost:8080/api/v1/user/login/authenticate',
+    body: {
+      password: testPassword,
+      username: committeeID + email,
+    },
+  }).then((resp) => {
+    if (resp.body.token) {
+      cy.setCookie('user', `%22${resp.body.token}%22`);
+      console.log('authenticate', resp);
```
Comment (dheitzer) [inline front-end/cypress/support/commands.ts]: maybe remove this

**Review by dheitzer — APPROVED** [review]

File: front-end/cypress/support/commands.ts (hunk 1 above)
Comment (Elaine-Krauss-TCG) [reply front-end/cypress/support/commands.ts]: :+1:

File: front-end/cypress/support/commands.ts (hunk 2 above)
Comment (Elaine-Krauss-TCG) [reply front-end/cypress/support/commands.ts]: :+1:

**Review by toddlees — APPROVED** [review]

**Pull request merged**